Author: Daniel Baumann <daniel.baumann@progress-technologies.net>
Description: Also removing auto symlink when destroying containers.

diff -Naurp lxc.orig/src/lxc/lxc-destroy.in lxc/src/lxc/lxc-destroy.in
--- lxc.orig/src/lxc/lxc-destroy.in
+++ lxc/src/lxc/lxc-destroy.in
@@ -130,3 +130,6 @@ fi
 
 # recursively remove the container to remove old container configuration
 rm -rf --one-file-system --preserve-root $lxc_path/$lxc_name
+
+# remove auto symlink
+rm -f /etc/lxc/auto/$lxc_name
