Author: Daniel Baumann <mail@daniel-baumann.ch>
Description: Correcting wrong toolname in lxc-unshare manpage (Closes: #705709).

diff -Naurp lxc.orig/doc/lxc-unshare.sgml.in lxc/doc/lxc-unshare.sgml.in
--- lxc.orig/doc/lxc-unshare.sgml.in
+++ lxc/doc/lxc-unshare.sgml.in
@@ -49,7 +49,7 @@ Foundation, Inc., 59 Temple Place, Suite
 
   <refsynopsisdiv>
     <cmdsynopsis>
-      <command>lxc-clone</command>
+      <command>lxc-unshare</command>
       <arg choice="req">-s <replaceable>namespaces</replaceable></arg>
       <arg choice="req">-u <replaceable>user</replaceable></arg>
       <arg choice="req">command</arg>
@@ -115,7 +115,7 @@ Foundation, Inc., 59 Temple Place, Suite
       <para>
         To spawn a new shell with its own UTS (hostname) namespace,
         <programlisting>
-          lxc-clone -s UTSNAME /bin/bash
+          lxc-unshare -s UTSNAME /bin/bash
         </programlisting>
 	If the hostname is changed in that shell, the change will not be
 	reflected on the host.
@@ -123,7 +123,7 @@ Foundation, Inc., 59 Temple Place, Suite
       <para>
         To spawn a shell in a new network, pid, and mount namespace,
         <programlisting>
-          lxc-clone -s "NETWORK|PID|MOUNT" /bin/bash
+          lxc-unshare -s "NETWORK|PID|MOUNT" /bin/bash
         </programlisting>
 	The resulting shell will have pid 1 and will see no network interfaces.
 	After re-mounting /proc in that shell,
