Author: Daniel Baumann <daniel.baumann@progress-technologies.net>
Description: Setting template directory to /usr/share/lxc/templates.

diff -Naurp lxc.orig/configure lxc/configure
--- lxc.orig/configure	2012-06-13 18:46:21.922951004 +0200
+++ lxc/configure	2012-06-14 08:14:05.717758441 +0200
@@ -4696,7 +4696,7 @@ LXCROOTFSMOUNT="${with_rootfs_path}"
 
 LXCINITDIR='${libexecdir}/lxc'
 
-LXCTEMPLATEDIR='${libdir}/lxc/templates'
+LXCTEMPLATEDIR='/usr/share/lxc/templates'
 
 
 for ac_header in linux/netlink.h linux/genetlink.h
diff -Naurp lxc.orig/configure.ac lxc/configure.ac
--- lxc.orig/configure.ac	2012-06-13 18:46:21.922951004 +0200
+++ lxc/configure.ac	2012-06-14 08:14:19.753853748 +0200
@@ -63,7 +63,7 @@ AS_AC_EXPAND(LXC_GENERATE_DATE, "$(date)
 AC_SUBST(LXCPATH, "${with_config_path}")
 AC_SUBST(LXCROOTFSMOUNT, "${with_rootfs_path}")
 AC_SUBST(LXCINITDIR, ['${libexecdir}/lxc'])
-AC_SUBST(LXCTEMPLATEDIR, ['${libdir}/lxc/templates'])
+AC_SUBST(LXCTEMPLATEDIR, ['/usr/share/lxc/templates'])
 
 AC_CHECK_HEADERS([linux/netlink.h linux/genetlink.h],
 				  [],
