Author: Daniel Baumann <daniel.baumann@progress-technologies.net>
Description: Avoid using bashisms in lxc commands.

diff -Naurp lxc.orig/src/lxc/lxc-clone.in lxc/src/lxc/lxc-clone.in
--- lxc.orig/src/lxc/lxc-clone.in	2012-11-12 13:28:25.720114813 +0100
+++ lxc/src/lxc/lxc-clone.in	2012-11-12 13:56:39.916515988 +0100
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/bin/sh
 
 #
 # lxc: linux Container library
diff -Naurp lxc.orig/src/lxc/lxc-destroy.in lxc/src/lxc/lxc-destroy.in
--- lxc.orig/src/lxc/lxc-destroy.in	2012-12-23 08:47:47.124062280 +0100
+++ lxc/src/lxc/lxc-destroy.in	2012-12-23 08:53:32.312142244 +0100
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/bin/sh
 
 #
 # lxc: linux Container library
@@ -111,13 +111,13 @@ fi
 # else, ignore it. We'll support deletion of others later.
 rootdev=`grep lxc.rootfs $lxc_path/$lxc_name/config 2>/dev/null | sed -e 's/^[^/]*/\//'`
 if [ -n "$rootdev" ]; then
-    if [ -b "$rootdev" -o -h "$rootdev" ]; then
+    if [ -b "$rootdev"] || [ -h "$rootdev" ]; then
         lvdisplay $rootdev > /dev/null 2>&1
         if [ $? -eq 0 ]; then
             echo "removing backing store: $rootdev"
             lvremove -f $rootdev
         fi
-    elif [ -h "$rootdev" -o -d "$rootdev" ]; then
+    elif [ -h "$rootdev"] || [ -d "$rootdev" ]; then
         if which btrfs >/dev/null 2>&1 &&
            btrfs subvolume list "$rootdev" >/dev/null 2>&1; then
             btrfs subvolume delete "$rootdev"
diff -Naurp lxc.orig/src/lxc/lxc-netstat.in lxc/src/lxc/lxc-netstat.in
--- lxc.orig/src/lxc/lxc-netstat.in	2012-12-23 08:54:49.560160061 +0100
+++ lxc/src/lxc/lxc-netstat.in	2012-12-23 08:59:22.276225392 +0100
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/bin/sh
 
 #
 # lxc: linux Container library
@@ -58,7 +58,7 @@ get_parent_cgroup()
 
         # Return the absolute path to the containers' parent cgroup
         # (do not append '/lxc' if the hierarchy contains the 'ns' subsystem)
-        if [[ ",$subsystems," == *,ns,* ]]; then
+        if [[ ",$subsystems," = *,ns,* ]]; then
             parent_cgroup="${mountpoint}${init_cgroup%/}"
         else
             parent_cgroup="${mountpoint}${init_cgroup%/}/lxc"
@@ -119,7 +119,7 @@ fi
 
 tmpdir=$(mktemp -d)
 
-if [ -z "$tmpdir" -o ! -d "$tmpdir" ]; then
+if [ -z "$tmpdir" ] || [ ! -d "$tmpdir" ]; then
     echo "$(basename $0): unable to create temporary directory" >&2
     exit 1
 fi
diff -Naurp lxc.orig/src/lxc/lxc-ps.in lxc/src/lxc/lxc-ps.in
--- lxc.orig/src/lxc/lxc-ps.in	2012-12-23 08:54:49.560160061 +0100
+++ lxc/src/lxc/lxc-ps.in	2012-12-23 09:02:14.708268697 +0100
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/bin/sh
 
 #
 # lxc: linux Container library
@@ -61,7 +61,7 @@ get_parent_cgroup()
 
         # Return the absolute path to the containers' parent cgroup
         # (do not append '/lxc' if the hierarchy contains the 'ns' subsystem)
-        if [[ ",$subsystems," == *,ns,* ]]; then
+        if [[ ",$subsystems," = *,ns,* ]]; then
             parent_cgroup="${mountpoint}${init_cgroup%/}"
         else
             parent_cgroup="${mountpoint}${init_cgroup%/}/lxc"
@@ -136,7 +136,7 @@ while read line; do
 
     buffer=${line:0:$line_pid_end_position}
     pid=${buffer##* }
-    if [ "$list_container_processes" -eq "0" -o ! -z "${container_of_pid[pid]}" ]; then
+    if [ "$list_container_processes" -eq "0" ] || [ ! -z "${container_of_pid[pid]}" ]; then
         printf "%-${container_field_width}s %s\n" "${container_of_pid[pid]}" "$line"
     fi
 done < <(ps "$@")
diff -Naurp lxc.orig/src/lxc/lxc-shutdown.in lxc/src/lxc/lxc-shutdown.in
--- lxc.orig/src/lxc/lxc-shutdown.in	2012-11-12 13:28:25.724114837 +0100
+++ lxc/src/lxc/lxc-shutdown.in	2012-11-12 13:56:57.316602038 +0100
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/bin/sh
 
 # (C) Copyright Canonical 2011,2012
 
