Author: Daniel Baumann <daniel.baumann@progress-technologies.net>
Description: Avoid using bashisms in lxc commands.

diff -Naurp lxc.orig/src/lxc/lxc-clone.in lxc/src/lxc/lxc-clone.in
--- lxc.orig/src/lxc/lxc-clone.in	2012-11-12 13:28:25.720114813 +0100
+++ lxc/src/lxc/lxc-clone.in	2012-11-12 13:56:39.916515988 +0100
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/bin/sh
 
 #
 # lxc: linux Container library
diff -Naurp lxc.orig/src/lxc/lxc-destroy.in lxc/src/lxc/lxc-destroy.in
--- lxc.orig/src/lxc/lxc-destroy.in	2012-11-12 13:28:25.720114813 +0100
+++ lxc/src/lxc/lxc-destroy.in	2012-11-12 13:56:41.796525204 +0100
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/bin/sh
 
 #
 # lxc: linux Container library
@@ -116,7 +116,7 @@ if [ -n "$rootdev" ]; then
 			echo "removing backing store: $rootdev"
 			lvremove -f $rootdev
 		fi
-	elif [ -h "$rootdev" -o -d "$rootdev" ]; then
+	elif [ -h "$rootdev" ] || [ -d "$rootdev" ]; then
 		if which btrfs >/dev/null 2>&1 &&
 		   btrfs subvolume list "$rootdev" >/dev/null 2>&1; then
 			btrfs subvolume delete "$rootdev"
diff -Naurp lxc.orig/src/lxc/lxc-ls.in lxc/src/lxc/lxc-ls.in
--- lxc.orig/src/lxc/lxc-ls.in	2012-11-12 13:28:25.720114813 +0100
+++ lxc/src/lxc/lxc-ls.in	2012-11-12 13:57:28.524756889 +0100
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/bin/sh
 
 #
 # lxc: linux Container library
@@ -61,7 +61,7 @@ get_parent_cgroup()
 
 		# Return the absolute path to the containers' parent cgroup
 		# (do not append '/lxc' if the hierarchy contains the 'ns' subsystem)
-		if [[ ",$subsystems," == *,ns,* ]]; then
+		if [ ",$subsystems," = *,ns,* ]; then
 			parent_cgroup="${mountpoint}${init_cgroup%/}"
 		else
 			parent_cgroup="${mountpoint}${init_cgroup%/}/lxc"
diff -Naurp lxc.orig/src/lxc/lxc-netstat.in lxc/src/lxc/lxc-netstat.in
--- lxc.orig/src/lxc/lxc-netstat.in	2012-11-12 13:28:25.720114813 +0100
+++ lxc/src/lxc/lxc-netstat.in	2012-11-12 13:59:03.261226608 +0100
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/bin/sh
 
 #
 # lxc: linux Container library
@@ -58,7 +58,7 @@ get_parent_cgroup()
 
 		# Return the absolute path to the containers' parent cgroup
 		# (do not append '/lxc' if the hierarchy contains the 'ns' subsystem)
-		if [[ ",$subsystems," == *,ns,* ]]; then
+		if [ ",$subsystems," = *,ns,* ]; then
 			parent_cgroup="${mountpoint}${init_cgroup%/}"
 		else
 			parent_cgroup="${mountpoint}${init_cgroup%/}/lxc"
@@ -119,7 +119,7 @@ fi
 
 tmpdir=$(mktemp -d)
 
-if [ -z "$tmpdir" -o ! -d "$tmpdir" ]; then
+if [ -z "$tmpdir" ] || [ ! -d "$tmpdir" ]; then
 	echo "$(basename $0): unable to create temporary directory" >&2
 	exit 1
 fi
diff -Naurp lxc.orig/src/lxc/lxc-ps.in lxc/src/lxc/lxc-ps.in
--- lxc.orig/src/lxc/lxc-ps.in	2012-11-12 13:28:25.720114813 +0100
+++ lxc/src/lxc/lxc-ps.in	2012-11-12 13:58:48.877155967 +0100
@@ -61,7 +61,7 @@ get_parent_cgroup()
 
 		# Return the absolute path to the containers' parent cgroup
 		# (do not append '/lxc' if the hierarchy contains the 'ns' subsystem)
-		if [[ ",$subsystems," == *,ns,* ]]; then
+		if [ ",$subsystems," = *,ns,* ]; then
 			parent_cgroup="${mountpoint}${init_cgroup%/}"
 		else
 			parent_cgroup="${mountpoint}${init_cgroup%/}/lxc"
@@ -136,7 +136,7 @@ while read line; do
 
 	buffer=${line:0:$line_pid_end_position}
 	pid=${buffer##* }
-	if [ "$list_container_processes" -eq "0" -o ! -z "${container_of_pid[pid]}" ]; then
+	if [ "$list_container_processes" -eq "0" ] || [ ! -z "${container_of_pid[pid]}" ]; then
 		printf "%-${container_field_width}s %s\n" "${container_of_pid[pid]}" "$line"
 	fi
 done < <(ps "$@")
diff -Naurp lxc.orig/src/lxc/lxc-setcap.in lxc/src/lxc/lxc-setcap.in
--- lxc.orig/src/lxc/lxc-setcap.in	2012-11-12 13:28:25.720114813 +0100
+++ lxc/src/lxc/lxc-setcap.in	2012-11-12 13:56:50.740569570 +0100
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/bin/sh
 
 #
 # lxc: linux Container library
diff -Naurp lxc.orig/src/lxc/lxc-setuid.in lxc/src/lxc/lxc-setuid.in
--- lxc.orig/src/lxc/lxc-setuid.in	2012-11-12 13:28:25.724114837 +0100
+++ lxc/src/lxc/lxc-setuid.in	2012-11-12 13:56:53.344582295 +0100
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/bin/sh
 
 #
 # lxc: linux Container library
diff -Naurp lxc.orig/src/lxc/lxc-shutdown.in lxc/src/lxc/lxc-shutdown.in
--- lxc.orig/src/lxc/lxc-shutdown.in	2012-11-12 13:28:25.724114837 +0100
+++ lxc/src/lxc/lxc-shutdown.in	2012-11-12 13:56:57.316602038 +0100
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/bin/sh
 
 # (C) Copyright Canonical 2011,2012
 
diff -Naurp lxc.orig/src/lxc/lxc-version.in lxc/src/lxc/lxc-version.in
--- lxc.orig/src/lxc/lxc-version.in	2012-11-12 13:28:23.488103745 +0100
+++ lxc/src/lxc/lxc-version.in	2012-11-12 13:57:01.784624211 +0100
@@ -1,3 +1,3 @@
-#!/bin/bash
+#!/bin/sh
 
 echo "lxc version: @PACKAGE_VERSION@"
