Author: Daniel Baumann <daniel.baumann@progress-technologies.net>
Description: Avoid using bashisms in lxc commands.

diff -Naurp lxc.orig/src/lxc/lxc-checkconfig.in lxc/src/lxc/lxc-checkconfig.in
--- lxc.orig/src/lxc/lxc-checkconfig.in	2012-11-12 13:28:25.720114813 +0100
+++ lxc/src/lxc/lxc-checkconfig.in	2012-11-12 13:57:20.456716739 +0100
@@ -1,37 +1,40 @@
-#!/bin/bash
+#!/bin/sh
 
 # Allow environment variables to override grep and config
 : ${CONFIG:=/proc/config.gz}
 : ${GREP:=zgrep}
 
-SETCOLOR_SUCCESS="echo -en \\033[1;32m"
-SETCOLOR_FAILURE="echo -en \\033[1;31m"
-SETCOLOR_WARNING="echo -en \\033[1;33m"
-SETCOLOR_NORMAL="echo -en \\033[0;39m"
+SETCOLOR_SUCCESS () { printf '\033[1;32m'; }
+SETCOLOR_FAILURE () { printf '\033[1;31m'; }
+SETCOLOR_WARNING () { printf '\033[1;33m'; }
+SETCOLOR_NORMAL  () { printf '\033[0;39m'; }
 
 is_set() {
-    $GREP -q "$1=[y|m]" $CONFIG
+    $GREP -q "$1=[y|m]" "$CONFIG"
     return $?
 }
 
 is_enabled() {
-    mandatory=$2
+    prefix=$1
+    option=$2
+    mandatory=$3
 
-    is_set $1
+    is_set "$option"
     RES=$?
 
+    echo -n "$prefix"
     if [ $RES -eq 0 ]; then
-	$SETCOLOR_SUCCESS && echo -e "enabled" && $SETCOLOR_NORMAL
+	SETCOLOR_SUCCESS && echo "enabled" && SETCOLOR_NORMAL
     else
-	if [ ! -z "$mandatory" -a "$mandatory" = yes ]; then
-	    $SETCOLOR_FAILURE && echo -e "required" && $SETCOLOR_NORMAL
-	else
-	    $SETCOLOR_WARNING && echo -e "missing" && $SETCOLOR_NORMAL
-	fi
-    fi
-}
-
-if [ ! -f $CONFIG ]; then
+	if [ ! -z "$mandatory" ] && [ "$mandatory" = yes ]; then
+	    SETCOLOR_FAILURE && echo "required" && SETCOLOR_NORMAL
+ 	else
+	    SETCOLOR_WARNING && echo "missing" && SETCOLOR_NORMAL
+ 	fi
+     fi
+ }
+ 
+if [ ! -f "$CONFIG" ]; then
     echo "Kernel configuration not found at $CONFIG; searching..."
     KVER="`uname -r`"
     HEADERS_CONFIG="/lib/modules/$KVER/build/.config"
@@ -52,13 +55,13 @@ if [ ! -f $CONFIG ]; then
 fi
 
 echo "--- Namespaces ---"
-echo -n "Namespaces: " && is_enabled CONFIG_NAMESPACES yes
-echo -n "Utsname namespace: " && is_enabled CONFIG_UTS_NS
-echo -n "Ipc namespace: " && is_enabled CONFIG_IPC_NS yes
-echo -n "Pid namespace: " && is_enabled CONFIG_PID_NS yes
-echo -n "User namespace: " && is_enabled CONFIG_USER_NS
-echo -n "Network namespace: " && is_enabled CONFIG_NET_NS
-echo -n "Multiple /dev/pts instances: " && is_enabled DEVPTS_MULTIPLE_INSTANCES
+is_enabled "Namespaces: " CONFIG_NAMESPACES yes
+is_enabled "Utsname namespace: " CONFIG_UTS_NS
+is_enabled "Ipc namespace: " CONFIG_IPC_NS yes
+is_enabled "Pid namespace: " CONFIG_PID_NS yes
+is_enabled "User namespace: " CONFIG_USER_NS
+is_enabled "Network namespace: " CONFIG_NET_NS
+is_enabled "Multiple /dev/pts instances: " DEVPTS_MULTIPLE_INSTANCES
 echo
 echo "--- Control groups ---"
 
@@ -69,42 +72,34 @@ print_cgroups() {
 
 CGROUP_MNT_PATH=`print_cgroups cgroup /proc/self/mounts | head -1`
 
-echo -n "Cgroup: " && is_enabled CONFIG_CGROUPS yes
+is_enabled "Cgroup: " CONFIG_CGROUPS yes
 
-if [ -f $CGROUP_MNT_PATH/cgroup.clone_children ]; then
+if [ -f "$CGROUP_MNT_PATH/cgroup.clone_children" ]; then
     echo -n "Cgroup clone_children flag: " &&
-    $SETCOLOR_SUCCESS && echo -e "enabled" && $SETCOLOR_NORMAL
+    SETCOLOR_SUCCESS && echo "enabled" && SETCOLOR_NORMAL
 else
     echo -n "Cgroup namespace: " && is_enabled CONFIG_CGROUP_NS yes
 fi
-echo -n "Cgroup device: " && is_enabled CONFIG_CGROUP_DEVICE
-echo -n "Cgroup sched: " && is_enabled CONFIG_CGROUP_SCHED
-echo -n "Cgroup cpu account: " && is_enabled CONFIG_CGROUP_CPUACCT
-echo -n "Cgroup memory controller: " && is_enabled CONFIG_CGROUP_MEM_RES_CTLR
-is_set CONFIG_SMP && echo -n "Cgroup cpuset: " && is_enabled CONFIG_CPUSETS
+is_enabled "Cgroup device: " CONFIG_CGROUP_DEVICE
+is_enabled "Cgroup sched: " CONFIG_CGROUP_SCHED
+is_enabled "Cgroup cpu account: " CONFIG_CGROUP_CPUACCT
+is_enabled "Cgroup memory controller: " CONFIG_CGROUP_MEM_RES_CTLR
+is_set CONFIG_SMP && is_enabled "Cgroup cpuset: " CONFIG_CPUSETS
 echo
 echo "--- Misc ---"
-echo -n "Veth pair device: " && is_enabled CONFIG_VETH
-echo -n "Macvlan: " && is_enabled CONFIG_MACVLAN
-echo -n "Vlan: " && is_enabled CONFIG_VLAN_8021Q
-KVER_MAJOR=$($GREP '^# Linux' $CONFIG | \
-    sed -r 's/.* ([0-9])\.[0-9]{1,2}\.[0-9]{1,3}.*/\1/')
-if [[ $KVER_MAJOR == 2 ]]; then
-KVER_MINOR=$($GREP '^# Linux' $CONFIG | \
-    sed -r 's/.* 2.6.([0-9]{2}).*/\1/')
+is_enabled "Veth pair device: " CONFIG_VETH
+is_enabled "Macvlan: " CONFIG_MACVLAN
+is_enabled "Vlan: " CONFIG_VLAN_8021Q
+KVER=$($GREP "^# Linux" "$CONFIG" | sed -r "s/.*([23])\.([0-9])+\.([0-9]+).*/\1 \2 \3/")
+kernel_version () { echo $(( ( $1 << 16 ) + ( $2 << 8 ) + $3)); }
+echo -n "File capabilities: "
+if [ $(kernel_version $KVER) -le $(kernel_version 2 6 32) ]; then
+        is_enabled CONFIG_SECURITY_FILE_CAPABILITIES
 else
-KVER_MINOR=$($GREP '^# Linux' $CONFIG | \
-    sed -r 's/.* [0-9]\.([0-9]{1,3})\.[0-9]{1,3}.*/\1/')
+        SETCOLOR_SUCCESS && echo "enabled" && SETCOLOR_NORMAL
 fi
-echo -n "File capabilities: " &&
-    ( [[ ${KVER_MAJOR} == 2 && ${KVER_MINOR} < 33 ]] &&
-       is_enabled CONFIG_SECURITY_FILE_CAPABILITIES ) ||
-    ( [[ ( ${KVER_MAJOR} == 2 && ${KVER_MINOR} > 32 ) ||
-         ${KVER_MAJOR} > 2 ]] && $SETCOLOR_SUCCESS &&
-         echo -e "enabled" && $SETCOLOR_NORMAL )
 
 echo
 echo "Note : Before booting a new kernel, you can check its configuration"
 echo "usage : CONFIG=/path/to/config $0"
 echo
-
diff -Naurp lxc.orig/src/lxc/lxc-clone.in lxc/src/lxc/lxc-clone.in
--- lxc.orig/src/lxc/lxc-clone.in	2012-11-12 13:28:25.720114813 +0100
+++ lxc/src/lxc/lxc-clone.in	2012-11-12 13:56:39.916515988 +0100
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/bin/sh
 
 #
 # lxc: linux Container library
diff -Naurp lxc.orig/src/lxc/lxc-create.in lxc/src/lxc/lxc-create.in
--- lxc.orig/src/lxc/lxc-create.in	2012-11-12 13:39:53.407524906 +0100
+++ lxc/src/lxc/lxc-create.in	2012-11-12 13:56:40.892520422 +0100
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/bin/sh
 
 #
 # lxc: linux Container library
@@ -139,7 +139,7 @@ done
 wantedhelp=0
 for var in "$@"
 do
-if [ "$var" = "-h" -o "$var" = "--help" ]; then
+if [ "$var" = "-h" ] || [ "$var" = "--help" ]; then
     help
     exit 1
 fi
@@ -186,7 +186,7 @@ fi
 
 rootfs="$lxc_path/$lxc_name/rootfs"
 
-if [ "$backingstore" = "_unset" -o "$backingstore" = "btrfs" ]; then
+if [ "$backingstore" = "_unset" ] || [ "$backingstore" = "btrfs" ]; then
 # if no backing store was given, then see if btrfs would work
     if which btrfs >/dev/null 2>&1 &&
         btrfs filesystem df "$lxc_path/" >/dev/null 2>&1; then
diff -Naurp lxc.orig/src/lxc/lxc-destroy.in lxc/src/lxc/lxc-destroy.in
--- lxc.orig/src/lxc/lxc-destroy.in	2012-11-12 13:28:25.720114813 +0100
+++ lxc/src/lxc/lxc-destroy.in	2012-11-12 13:56:41.796525204 +0100
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/bin/sh
 
 #
 # lxc: linux Container library
@@ -116,7 +116,7 @@ if [ -n "$rootdev" ]; then
 			echo "removing backing store: $rootdev"
 			lvremove -f $rootdev
 		fi
-	elif [ -h "$rootdev" -o -d "$rootdev" ]; then
+	elif [ -h "$rootdev" ] || [ -d "$rootdev" ]; then
 		if which btrfs >/dev/null 2>&1 &&
 		   btrfs subvolume list "$rootdev" >/dev/null 2>&1; then
 			btrfs subvolume delete "$rootdev"
diff -Naurp lxc.orig/src/lxc/lxc-ls.in lxc/src/lxc/lxc-ls.in
--- lxc.orig/src/lxc/lxc-ls.in	2012-11-12 13:28:25.720114813 +0100
+++ lxc/src/lxc/lxc-ls.in	2012-11-12 13:57:28.524756889 +0100
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/bin/sh
 
 #
 # lxc: linux Container library
@@ -61,7 +61,7 @@ get_parent_cgroup()
 
 		# Return the absolute path to the containers' parent cgroup
 		# (do not append '/lxc' if the hierarchy contains the 'ns' subsystem)
-		if [[ ",$subsystems," == *,ns,* ]]; then
+		if [ ",$subsystems," = *,ns,* ]; then
 			parent_cgroup="${mountpoint}${init_cgroup%/}"
 		else
 			parent_cgroup="${mountpoint}${init_cgroup%/}/lxc"
diff -Naurp lxc.orig/src/lxc/lxc-netstat.in lxc/src/lxc/lxc-netstat.in
--- lxc.orig/src/lxc/lxc-netstat.in	2012-11-12 13:28:25.720114813 +0100
+++ lxc/src/lxc/lxc-netstat.in	2012-11-12 13:59:03.261226608 +0100
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/bin/sh
 
 #
 # lxc: linux Container library
@@ -58,7 +58,7 @@ get_parent_cgroup()
 
 		# Return the absolute path to the containers' parent cgroup
 		# (do not append '/lxc' if the hierarchy contains the 'ns' subsystem)
-		if [[ ",$subsystems," == *,ns,* ]]; then
+		if [ ",$subsystems," = *,ns,* ]; then
 			parent_cgroup="${mountpoint}${init_cgroup%/}"
 		else
 			parent_cgroup="${mountpoint}${init_cgroup%/}/lxc"
@@ -119,7 +119,7 @@ fi
 
 tmpdir=$(mktemp -d)
 
-if [ -z "$tmpdir" -o ! -d "$tmpdir" ]; then
+if [ -z "$tmpdir" ] || [ ! -d "$tmpdir" ]; then
 	echo "$(basename $0): unable to create temporary directory" >&2
 	exit 1
 fi
diff -Naurp lxc.orig/src/lxc/lxc-ps.in lxc/src/lxc/lxc-ps.in
--- lxc.orig/src/lxc/lxc-ps.in	2012-11-12 13:28:25.720114813 +0100
+++ lxc/src/lxc/lxc-ps.in	2012-11-12 13:58:48.877155967 +0100
@@ -61,7 +61,7 @@ get_parent_cgroup()
 
 		# Return the absolute path to the containers' parent cgroup
 		# (do not append '/lxc' if the hierarchy contains the 'ns' subsystem)
-		if [[ ",$subsystems," == *,ns,* ]]; then
+		if [ ",$subsystems," = *,ns,* ]; then
 			parent_cgroup="${mountpoint}${init_cgroup%/}"
 		else
 			parent_cgroup="${mountpoint}${init_cgroup%/}/lxc"
@@ -136,7 +136,7 @@ while read line; do
 
 	buffer=${line:0:$line_pid_end_position}
 	pid=${buffer##* }
-	if [ "$list_container_processes" -eq "0" -o ! -z "${container_of_pid[pid]}" ]; then
+	if [ "$list_container_processes" -eq "0" ] || [ ! -z "${container_of_pid[pid]}" ]; then
 		printf "%-${container_field_width}s %s\n" "${container_of_pid[pid]}" "$line"
 	fi
 done < <(ps "$@")
diff -Naurp lxc.orig/src/lxc/lxc-setcap.in lxc/src/lxc/lxc-setcap.in
--- lxc.orig/src/lxc/lxc-setcap.in	2012-11-12 13:28:25.720114813 +0100
+++ lxc/src/lxc/lxc-setcap.in	2012-11-12 13:56:50.740569570 +0100
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/bin/sh
 
 #
 # lxc: linux Container library
diff -Naurp lxc.orig/src/lxc/lxc-setuid.in lxc/src/lxc/lxc-setuid.in
--- lxc.orig/src/lxc/lxc-setuid.in	2012-11-12 13:28:25.724114837 +0100
+++ lxc/src/lxc/lxc-setuid.in	2012-11-12 13:56:53.344582295 +0100
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/bin/sh
 
 #
 # lxc: linux Container library
diff -Naurp lxc.orig/src/lxc/lxc-shutdown.in lxc/src/lxc/lxc-shutdown.in
--- lxc.orig/src/lxc/lxc-shutdown.in	2012-11-12 13:28:25.724114837 +0100
+++ lxc/src/lxc/lxc-shutdown.in	2012-11-12 13:56:57.316602038 +0100
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/bin/sh
 
 # (C) Copyright Canonical 2011,2012
 
diff -Naurp lxc.orig/src/lxc/lxc-version.in lxc/src/lxc/lxc-version.in
--- lxc.orig/src/lxc/lxc-version.in	2012-11-12 13:28:23.488103745 +0100
+++ lxc/src/lxc/lxc-version.in	2012-11-12 13:57:01.784624211 +0100
@@ -1,3 +1,3 @@
-#!/bin/bash
+#!/bin/sh
 
 echo "lxc version: @PACKAGE_VERSION@"
