Author: Daniel Baumann <daniel.baumann@progress-technologies.net>
Description: Don't list vendor specific example template.

diff -Naurp lxc.orig/src/lxc/lxc-create.in lxc/src/lxc/lxc-create.in
--- lxc.orig/src/lxc/lxc-create.in	2012-12-23 09:08:25.512359153 +0100
+++ lxc/src/lxc/lxc-create.in	2012-12-23 09:09:59.104381579 +0100
@@ -51,7 +51,7 @@ help() {
     echo >&2
     if [ -z "$lxc_template" ]; then
         echo "To see template-specific options, specify a template. For example:" >&2
-        echo "  $(basename $0) -t ubuntu -h" >&2
+        echo "  $(basename $0) -t debconfig -h" >&2
         exit 0
     fi
     if [ -x ${templatedir}/lxc-$lxc_template ]; then
