Author: Serge Hallyn <serge.hallyn@ubuntu.com>
Description: dont fail on failure to link kmsg

diff -Naurp lxc.orig/src/lxc/conf.c lxc/src/lxc/conf.c
--- lxc.orig/src/lxc/conf.c	2012-11-23 10:29:42.008080821 +0100
+++ lxc/src/lxc/conf.c	2012-11-26 20:16:04.709929941 +0100
@@ -2196,10 +2196,8 @@ int lxc_setup(const char *name, struct l
 		return -1;
 	}
 
-	if (setup_kmsg(&lxc_conf->rootfs, &lxc_conf->console)) {
+	if (setup_kmsg(&lxc_conf->rootfs, &lxc_conf->console))  // don't fail
 		ERROR("failed to setup kmsg for '%s'", name);
-		return -1;
-	}
 
 	if (setup_tty(&lxc_conf->rootfs, &lxc_conf->tty_info, lxc_conf->ttydir)) {
 		ERROR("failed to setup the ttys for '%s'", name);
