Author: Daniel Baumann <daniel.baumann@progress-technologies.net>
Description: Correct paths to lxc-init (Closes: #682790).

diff -Naurp lxc.orig/src/lxc/lxc-setcap.in lxc/src/lxc/lxc-setcap.in
--- lxc.orig/src/lxc/lxc-setcap.in	2012-07-26 01:51:06.383661274 +0200
+++ lxc/src/lxc/lxc-setcap.in	2012-07-26 01:50:50.675587139 +0200
@@ -52,7 +52,7 @@ lxc_setcaps()
     setcap $LXC_UNSHARE_CAPS=ep @BINDIR@/lxc-unshare
     setcap $LXC_NETSTAT_CAPS=ep @BINDIR@/lxc-netstat
     setcap $LXC_CHECKPOINT_CAPS=ep @BINDIR@/lxc-checkpoint
-    setcap $LXC_INIT_CAPS=ep @LXCINITDIR@/lxc-init
+    setcap $LXC_INIT_CAPS=ep @LIBDIR@/lxc-init
 
     test -e @LXCPATH@ || mkdir -p @LXCPATH@
     chmod 0777 @LXCPATH@
@@ -68,7 +68,7 @@ lxc_dropcaps()
     setcap -r @BINDIR@/lxc-unshare
     setcap -r @BINDIR@/lxc-netstat
     setcap -r @BINDIR@/lxc-checkpoint
-    setcap -r @LXCINITDIR@/lxc-init
+    setcap -r @LIBDIR@/lxc-init
     chmod 0755 @LXCPATH@
 }
 
